RecoLocalTracker/SiStripClusterizer/plugins/SiStripApprox2ApproxClusters.cc

```diff
@@ -0,0 +1,90 @@
+
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "DataFormats/SiStripCluster/interface/SiStripApproximateCluster.h"
+#include "DataFormats/SiStripCluster/interface/SiStripCluster.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+
+#include <vector>
+#include <memory>
+
+class SiStripApprox2ApproxClusters : public edm::stream::EDProducer<> {
+public:
+  explicit SiStripApprox2ApproxClusters(const edm::ParameterSet& conf);
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  edm::InputTag inputApproxClusters;
+  std::string approxVersion;
+  edm::EDGetTokenT<edmNew::DetSetVector<SiStripApproximateCluster>> clusterToken;
+};
+
+SiStripApprox2ApproxClusters::SiStripApprox2ApproxClusters(const edm::ParameterSet& conf) {
+  inputApproxClusters = conf.getParameter<edm::InputTag>("inputApproxClusters");
+  approxVersion = conf.getParameter<std::string>("approxVersion");
+
+  clusterToken = consumes<edmNew::DetSetVector<SiStripApproximateCluster>>(inputApproxClusters);
+  produces<edmNew::DetSetVector<SiStripApproximateCluster>>();
+}
+
+void SiStripApprox2ApproxClusters::produce(edm::Event& e, edm::EventSetup const&) {
+  auto result = std::make_unique<edmNew::DetSetVector<SiStripApproximateCluster>>();
+  edm::Handle<edmNew::DetSetVector<SiStripApproximateCluster>> clusterCollection = e.getHandle(clusterToken);
+
+  for (const auto& detClusters : *clusterCollection) {
+    std::vector<SiStripApproximateCluster> tempVec;
+    edmNew::DetSetVector<SiStripApproximateCluster>::FastFiller ff{*result, detClusters.id()};
+
+    for (const auto& cluster : detClusters) {
+      float barycenter = cluster.barycenter();
+      uint8_t width = cluster.width();
+      float avgCharge = cluster.avgCharge();
+
+      if (approxVersion == "ORIGINAL") {
```

Comment:
I'm worried about the cost of string comparisons here inside a `produce` method. Did you consider an alternative, e.g. enum?

Reply:
Ok, I removed the selector on the string at the level of constructor. In principle I could also pass directly a number as a parameter but for readability on the .py file, probably is better to keep a string. Please advise if otherwise. A new version of the code is being committed